com.unity.render-pipelines.universal/Editor/Camera/UniversalRenderPipelineCameraStackUI.cs

```diff
@@ -0,0 +1,296 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework;
+using UnityEditor.SceneManagement;
+using UnityEditorInternal;
+using UnityEngine;
+using UnityEngine.Rendering.Universal;
+
+namespace UnityEditor.Rendering.Universal
+{
+    internal class UniversalRenderPipelineCameraStackUI
+    {
+        static readonly List<CameraRenderType> s_ValidCameraTypes = new List<CameraRenderType>
+        {
+            CameraRenderType.Overlay
+            // Add here the supported types of cameras
+        };
+
+        static class Styles
+        {
+            public const int cameraNameWidth = 150;
+            public const int errorIconWidth = 12;
+
+            public static readonly Vector2 errorIconSize = new Vector2(errorIconWidth, errorIconWidth);
+            public static readonly GUIStyle errorStyle = new GUIStyle(EditorStyles.label) { padding = new RectOffset { left = -errorIconWidth }};
+
+            static readonly Texture2D errorIcon = EditorGUIUtility.Load("icons/console.erroricon.sml.png") as Texture2D;
+            public static readonly Dictionary<CameraRenderType, GUIContent> typeNotSupportedMessage = Enum.GetValues(typeof(CameraRenderType))
+                .Cast<CameraRenderType>()
+                .Where(cameraRenderType => !s_ValidCameraTypes.Contains(cameraRenderType))
+                .ToDictionary(cameraRenderType => cameraRenderType, cameraRenderType => EditorGUIUtility.TrTextContent(string.Empty, cameraRenderType.GetName() + " is not supported", errorIcon));
+
+            public static readonly GUIContent[] camerasToAddNotFound = { EditorGUIUtility.TrTextContent("Could not find suitable cameras") };
+            public static readonly GUIContent PP = EditorGUIUtility.TrTextContent("PP", "The camera has enabled the Post Processing");
+
+            public static readonly string validTypes = $"Valid types are: {string.Join(",", s_ValidCameraTypes.Select(i => $"{i}"))}";
+        }
+
+        readonly Camera m_Camera;
+        readonly ReorderableList m_LayerList;
+        readonly UniversalRenderPipelineSerializedCamera m_SerializedCamera;
+
+        bool CanRemove(ReorderableList list)
+        {
+            // As the list can delete the last item, allow the delete action if there is at least one item
+            return m_SerializedCamera.cameras.arraySize > 0;
+        }
+
+        /// <summary>
+        /// Returns a <see cref="Camera"/> at a given index
+        /// </summary>
+        /// <param name="index">The index to fetch the camera</param>
+        /// <returns><see cref="Camera"/> if the index is valid, otherwise returns null</returns>
+        public Camera this[int index]
+        {
+            get
+            {
+                if (index < 0 || index > m_SerializedCamera.cameras.arraySize - 1)
```

Comment:
I have a preference for writing 
`index >= m_SerializedCamera.cameras.arraySize`
instead of 
`index > m_SerializedCamera.cameras.arraySize - 1`